lib/portlayer/exec/container.go

```diff
@@ -451,8 +452,21 @@ func (c *Container) Remove(ctx context.Context, sess *session.Session) error {
 		return c.vm.DeleteExceptDisks(ctx)
 	})
 	if err != nil {
-		c.State = existingState
-		return err
+		f, ok := err.(types.HasFault)
+		if !ok {
+			c.State = existingState
+			return err
+		}
+		switch f.Fault().(type) {
+		case *types.InvalidState:
+			log.Warnf("container VM is in invalid state, unregistering")
+			if err := c.vm.Unregister(ctx); err != nil {
+				log.Errorf("Error while attempting to unregister container VM: %s", err)
+				return err
```

Comment:
There's no logic preventing attempted removal of a container whose current state is `StateRemoving`. Would seeing the container state as `Removing` in the output of `docker ps` be useful? If we reset it to `existingState`, it may not be evident that something went wrong (or is wrong with the container). What do you think?